server/tests/e2e.rs

```diff
@@ -176,6 +184,80 @@ async fn e2e_two_nodes_quote_propagation(logger: Logger) {
     assert_eq!(quote.sci(), &sci);
 }
 
+/// Test that nodes don't accept dust quotes directly, but will propagate dust
+/// quotes to each other.
+#[async_test_with_logger(flavor = "multi_thread")]
+async fn e2e_two_nodes_dust_propagation(logger: Logger) {
+    let mut rng: StdRng = SeedableRng::from_seed([1u8; 32]);
+    let ledger_db = create_and_initialize_test_ledger();
+
+    // Start two DEQS servers
+    let (deqs_server1, quote_book1, client1) = start_deqs_server(
+        &ledger_db,
+        &[],
+        &[],
+        vec![(TokenId::from(1), 1000)],
+        &logger,
+    )
+    .await;
+
+    let (_deqs_server2, _quote_book2, client2) = start_deqs_server(
+        &ledger_db,
+        &[&deqs_server1],
+        &[],
+        vec![(TokenId::from(1), 100)],
+        &logger,
+    )
+    .await;
+
+    // Submit an SCI to the first server
+    let dust_sci = deqs_mc_test_utils::create_sci(
+        TokenId::from(1),
+        TokenId::from(2),
+        500,
+        20000,
+        &mut rng,
+        Some(&ledger_db),
+    );
+
+    let req = SubmitQuotesRequest {
+        quotes: vec![(&dust_sci).into()].into(),
+        ..Default::default()
+    };
+
+    // Submitting to the first server should fail because it's smaller than its dust
+    // level.
+    let resp = client1.submit_quotes(&req).expect("submit quote failed");
+    assert_eq!(
+        resp.get_status_codes(),
+        vec![QuoteStatusCode::UNSUPPORTED_SCI]
+    );
+    assert_eq!(
+        resp.get_error_messages(),
+        vec!["Unsupported SCI: Quote volume is too small for deqs. Quotes with base_token: 1 require a minimum of: 1000"]
+    );
+
+    // Submitting to the second server should succeed because it's bigger than its
+    // dust level.
+    let resp = client2.submit_quotes(&req).expect("submit quote failed");
+    let quote_id = QuoteId::try_from(resp.get_quotes()[0].get_id()).unwrap();
+
+    // After a few moments the first server should have the SCI in its quote book
```

Comment:
```suggestion
    // After a few moments the first server should have the SCI in its quote book since peer to peer
    // quotes are exempt from the dust check.
```